src/mixins/java/org/spongepowered/common/mixin/core/server/players/PlayerListMixin.java

```diff
@@ -521,56 +509,56 @@ public abstract class PlayerListMixin implements PlayerListBridge {
             to = @At(value = "INVOKE", target = "Lnet/minecraft/server/MinecraftServer;isDemo()Z")
         )
     )
-    private boolean impl$flagIfRespawnLocationIsGameMechanic(final Optional<?> optional) {
-        this.impl$isGameMechanicRespawn = optional.isPresent();
-        return this.impl$isGameMechanicRespawn;
+    private boolean impl$flagIfRespawnPositionIsGameMechanic(final Optional<Vec3> respawnPosition) {
+        this.impl$isGameMechanicRespawn = respawnPosition.isPresent();
+        return false; // force call of MinecraftServer#overworld which is redirected into impl$callRespawnPlayerSelectWorld
     }
 
-    @Redirect(method = "respawn",
-        at = @At(
-            value = "INVOKE",
-            target = "Lnet/minecraft/server/network/ServerGamePacketListenerImpl;send(Lnet/minecraft/network/protocol/Packet;)V",
-            ordinal = 1
-        )
-    )
-    private void impl$callRespawnPlayerRecreateEvent(
-        final ServerGamePacketListenerImpl serverPlayNetHandler, final Packet<?> packetIn, final net.minecraft.server.level.ServerPlayer originalPlayer, final boolean keepAllPlayerData) {
-        final net.minecraft.server.level.ServerPlayer recreatedPlayer = serverPlayNetHandler.player;
+    @Redirect(method = "respawn", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/MinecraftServer;overworld()Lnet/minecraft/server/level/ServerLevel;"))
```

Comment:
Just to be sure, is this only one call in `respawn`? Or are there multiple `overworld` calls?

Reply:
There is only one.